### PR TITLE
Tests on minikube

### PR DESCRIPTION
**Pull request by joan38**
Thanks @novakov-alexey for [suggesting this](https://github.com/joan38/kubernetes-client/pull/9#issuecomment-612872305). I managed to implement it with Github Actions.

Also we now have a link from the client libraries:
https://kubernetes.io/docs/reference/using-api/client-libraries/#community-maintained-client-libraries

### CODE REVIEW

**Comment by joan38** [comment]
For some reason the watch tests fails randomly on CI but work fine locally on my laptop with the same Kubernetes version (FYI I used 1.17.5 because 1.18.x [has an issue](https://github.com/manusa/actions-setup-minikube/issues/7)).
Any idea @novakov-alexey?

File: src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala [hunk 1]
```diff
@@ -48,39 +48,39 @@ trait WatchableTests[F[_], Resource <: { def metadata: Option[ObjectMeta] }]
...
-      val a = set.headOption.getOrElse(fail("stream should have at least one element with all received events"))
-      a should === (expected)
-    }
+    val watchEvents = for {
+      set <- watchStream.interruptAfter(60.seconds).compile.toList
+      result = set.headOption.getOrElse(fail("stream should have at least one element with all received events"))
+      _      = result should === (expected)
+    } yield ()
+
+    List(watchEvents, timer.sleep(1.second) *> sendEvents).parSequence
```
Comment (novakov-alexey-zz) [inline src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala]: Is this 1 second delay really needed here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,7 +12,17 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
-      - uses: olafurpg/setup-scala@v7
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup SBT
+        uses: olafurpg/setup-scala@v7
       - name: Style checks
         run: sbt styleCheck
+      - name: Setup Minikube and start Kubernetes v1.17.5
```
Comment (novakov-alexey-zz) [inline .github/workflows/ci.yml]: Cool that GitHub Actions is supporting Minikube as well.

**Review by novakov-alexey-zz — COMMENTED** [review]
Watchable Test may fail if client did not receive all expected events within the timeout (60 seconds). If one runs minikube on very weak/limited machine in terms of CPU/RAM, then it will be really slow when sending events to watchers. Same applies to all other operations. I don't really like this Watchable test, because of having that timeout. However, I could not find any other strategy while designing this test.

**Comment by novakov-alexey-zz** [comment]
> For some reason the watch tests fails randomly on CI but work fine locally on my laptop with the same Kubernetes version (FYI I used 1.17.5 because 1.18.x [has an issue](https://github.com/manusa/actions-setup-minikube/issues/7)).
> Any idea @novakov-alexey?

As I mentioned above. It may fail due to no events received within the expected timeout - 60 seconds. 

```
2020-04-19T04:57:56.5049022Z [info] - should watch a ConfigMap events *** FAILED ***
2020-04-19T04:57:56.5049543Z [info]   Set() did not equal Set(ADDED, MODIFIED, DELETED) (WatchableTests.scala:84)
```
Looking at above error from the CI logs, we did not receive all events from Minikube. We can try to set longer timeout just to see if it helps. Maybe to 5 minutes or so.

In my case, I use minikube and kubernetes:
😄  minikube v1.9.1 on Darwin 10.15.3
🐳  Preparing Kubernetes v1.18.0 on Docker 19.03.2 ...

File: src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala (hunk 1 above)
Comment (joan38) [reply src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala]: That's what made it pass. The events where sent ahead of the watch start.

**Comment by joan38** [comment]
All good, now we are testing against K8s 1.16 and 1.17.
Will add 1.18 once [the issue](https://github.com/manusa/actions-setup-minikube/issues/7) is resolved.
Thanks

File: src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala (hunk 1 above)
Comment (novakov-alexey-zz) [reply src/test/scala/com/goyeau/kubernetes/client/operation/WatchableTests.scala]: Interesting. I thought etcd always sends all events occured within the last 5 minutes.

**Comment by novakov-alexey-zz** [comment]
Nice. I believe testing against K8s 1.16.-1.17 is sufficient for now.

**Pull request merged**